File: src/main/java/uk/laxd/dndSimulator/event/SimpleEventLogger.java
package uk.laxd.dndSimulator.event;

import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.Collection;

@Component
@Scope("singleton")
public class SimpleEventLogger implements EventLogger {

    private int index = 1;

    private Collection<EncounterEvent> events = new ArrayList<>();

    public void logEvent(EncounterEventType encounterEventType) {
        EncounterEvent event = new EncounterEvent();
        event.type = encounterEventType;

        logEvent(event);
    }

    public void logEvent(EncounterEvent event) {
        // Set the index.
        event.setIndex(index);

        this.events.add(event);

        index++;
    }

    public Collection<EncounterEvent> getEvents() {
        return events;
    }

}

File: app/src/main/kotlin/uk/laxd/dndSimulator/event/SimpleEventLogger.kt
package uk.laxd.dndSimulator.event

import org.springframework.context.annotation.Scope
import org.springframework.stereotype.Component
import uk.laxd.dndSimulator.character.Character

@Component
@Scope("singleton")
class SimpleEventLogger : EventLogger {
    override val events: MutableList<EncounterEvent> = mutableListOf()

    override fun logEvent(actor: Character, eventType: EncounterEventType) {
        logEvent(GeneralEncounterEvent(actor, eventType))
    }

    override fun logEvent(event: EncounterEvent) {
        events.add(event)
    }
}